### PR TITLE
Add invalid put test that make 500 error in server

### PR DESCRIPTION
**Pull request by u5surf**
#112 
I also added some tests for which increase test coverage, made it indicated to be internal server error when PUT request in server.go

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -247,3 +248,43 @@ func TestCacheIndexHandler(t *testing.T) {
...
+
+	putCacheHandler(rr, req)
+	resp := rr.Result()
+
+	if resp.StatusCode != 500 {
+		t.Errorf("want: 500; got: %d", resp.StatusCode)
+	}
+}
+
+func blob(char byte, len int) []byte {
```
Comment (cristaloleg) [inline server/server_test.go]: https://godoc.org/bytes#Repeat

File: server/server_test.go [hunk 2]
```diff
@@ -247,3 +248,43 @@ func TestCacheIndexHandler(t *testing.T) {
...
+	b := make([]byte, len)
+	for index := range b {
+		b[index] = char
+	}
+	return b
+}
+
+type errReader int
+
+func (errReader) Read(p []byte) (n int, err error) {
```
Comment (cristaloleg) [inline server/server_test.go]: Due to dropped variable for `errReader` everything else can be also dropped 😉 
```suggestion
func (errReader) Read([]byte) (int, error) {
```

File: server/server_test.go (hunk 1 above)
Comment (u5surf) [reply server/server_test.go]: awesome , I think all blob function should be replaced to bytes.Repeat.

**Comment by mxplusb** [comment]
@u5surf can you please sign d42f76f or squash all of the commits so it can be merged? It looks good, just waiting on that.

**Pull request merged**